diff -ruN FD-3.01j.orig/dosdisk.c FD-3.01j/dosdisk.c
--- FD-3.01j.orig/dosdisk.c	2019-07-27 00:00:00.000000000 +0900
+++ FD-3.01j/dosdisk.c	2022-10-28 14:43:12.000000000 +0900
@@ -22,7 +22,7 @@
 #ifdef	LINUX
 #include "mntinfo.h"
 #include <sys/mount.h>
-#include <linux/unistd.h>
+#include <unistd.h>
 # ifndef	BLKFLSBUF
 # include <linux/fs.h>
 # endif
diff -ruN FD-3.01j.orig/machine.h FD-3.01j/machine.h
--- FD-3.01j.orig/machine.h	2019-07-27 00:00:00.000000000 +0900
+++ FD-3.01j/machine.h	2022-10-28 15:07:09.000000000 +0900
@@ -562,7 +562,7 @@
 # if	defined (Slackware)
 # define	TERMCAPLIB	"-ltermcap"
 # else
-# define	TERMCAPLIB	"-lncurses"
+# define	TERMCAPLIB	"-lcurses -lterminfo"
 # endif
 #define	DECLSIGLIST
 #define	DECLERRLIST
@@ -968,7 +968,9 @@
 
 /* #define CODEEUC	;kanji code type is EUC */
 /* #define DEFKCODE	;default string for kanji code type */
+#define DEFKCODE "utf8-iconv"
 /* #define UTF8DOC	;kanji code type for documents is UTF-8 */
+#define UTF8DOC
 /* #define UTF8LANG	;kanji code type if LANG environment value is UTF-8 */
 /* #define NOMULTIKANJI	;no need to support multiple kanji codes */
 /* #define PATHNOCASE	;pathname is case insensitive */
